plugins/autosuspend.koplugin/main.lua

```diff
@@ -49,15 +56,13 @@ function AutoSuspend:_schedule(shutdown_only)
 
     local delay_suspend, delay_shutdown
 
-    if PluginShare.pause_auto_suspend or Device.standby_prevented or Device.powerd:isCharging() then
+    if PluginShare.pause_auto_suspend or Device.powerd:isCharging() then
         delay_suspend = self.auto_suspend_timeout_seconds
         delay_shutdown = self.autoshutdown_timeout_seconds
     else
-        local now_tv = UIManager:getTime()
-        delay_suspend = self.last_action_tv + TimeVal:new{ sec = self.auto_suspend_timeout_seconds, usec = 0 } - now_tv
-        delay_suspend = delay_suspend:tonumber()
-        delay_shutdown = self.last_action_tv + TimeVal:new{ sec = self.autoshutdown_timeout_seconds, usec = 0 } - now_tv
-        delay_shutdown = delay_shutdown:tonumber()
+        local now = os.time()
```

Comment:
I'd rather keep using a monotonic clock source here, too.

(so, either figure out a way for UIManager's getTime to do the trick, or BOOTTIME w/ a REALTIME fallback).